### PR TITLE
Core/net6

### PR DESCRIPTION
**Pull request by michaelongso**
This is part of the great NET6 upgrade.
Build is updated to Nuke.
Pipeline is now using GitHub Actions.
Tests are now against Windows and Linux agents.
Package is now pushed to Octopus (which controls the release feed)

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -17,7 +17,6 @@
 [Rr]eleases/
 x64/
 x86/
-build/
```
Comment (michaelongso) [inline .gitignore:20]: Nuke src is under build/

File: .nuke/build.schema.json [hunk 2]
```diff
@@ -0,0 +1,120 @@
+{
```
Comment (michaelongso) [inline .nuke/build.schema.json:1]: Generated by Nuke

File: build.cmd [hunk 3]
```diff
@@ -1,25 +1,7 @@
...
-
-SET SCRIPTNAME=%~d0%~p0%~n0.ps1
-SET ARGS=%*
-IF [%1] NEQ [] GOTO ESCAPE_ARGS
-
-:POWERSHELL
-ECHO %ARGS%
-PowerShell.exe -NoProfile -NonInteractive -NoLogo -ExecutionPolicy Unrestricted -Command "& { $ErrorActionPreference = 'Stop'; & '%SCRIPTNAME%' @args; EXIT $LASTEXITCODE }" %ARGS%
-EXIT /B %ERRORLEVEL%
+:; set -eo pipefail
```
Comment (michaelongso) [inline build.cmd:1]: Updated by Nuke

File: build/.editorconfig [hunk 4]
```diff
@@ -0,0 +1,11 @@
+[*.cs]
```
Comment (michaelongso) [inline build/.editorconfig:1]: Generated by Nuke

File: global.json [hunk 5]
```diff
@@ -0,0 +1,6 @@
+{
```
Comment (michaelongso) [inline global.json:1]: Added for consistency with other Octopus repo

File: source/JavaPropertiesParser.Tests/JavaPropertiesParser.Tests.csproj [hunk 6]
```diff
@@ -1,15 +1,15 @@
...
+        <TargetFramework>net6.0</TargetFramework>
 
         <IsPackable>false</IsPackable>
     </PropertyGroup>
 
     <ItemGroup>
         <PackageReference Include="Assent" Version="1.6.1" />
-        <PackageReference Include="Fastenshtein" Version="1.0.0.5" />
-        <PackageReference Include="FluentAssertions" Version="5.10.3" />
+        <PackageReference Include="Fastenshtein" Version="1.0.0.8" />
```
Comment (michaelongso) [inline source/JavaPropertiesParser.Tests/JavaPropertiesParser.Tests.csproj:11]: Version upgrade are not really needed. Included during development. Didn't remove as all tests are still passing with it.

File: source/JavaPropertiesParser.Tests/ParserFixture.cs [hunk 7]
```diff
@@ -350,37 +351,49 @@ public void CanParseAValueWithALogicalNewLine()
         }
 
         [Test]
-        public void CanParseAValueWithAPhysicalNewLine()
+        [TestCase("value-with-physical-crlf-newline.properties")]
```
Comment (michaelongso) [inline source/JavaPropertiesParser.Tests/ParserFixture.cs:354]: Original test file is duplicated and updated where 1 file with CRLF (Windows) and the other with LF (Linux / MacOS)

File: source/JavaPropertiesParser.Tests/ParserFixture.cs [hunk 8]
```diff
@@ -350,37 +351,49 @@ public void CanParseAValueWithALogicalNewLine()
...
+            {
+                Value("value", "val\\\r\nue"), // Windows newline - CRLF.
+                Value("value", "val\\\nue") // Linux & MacOS newline - LF. 
+            };
+
+            parsed.Expressions.Should().ContainSingle().Which.Should().BeAssignableTo<KeyValuePairExpression>();
+            var resultExpression = (KeyValuePairExpression)parsed.Expressions.First();
+            resultExpression.Key.Should().Be(expectedKey);
+            resultExpression.Separator.Should().Be(expectedSeparator);
+            expectedValues.Should().Contain(resultExpression.Value);
```
Comment (michaelongso) [inline source/JavaPropertiesParser.Tests/ParserFixture.cs:373]: OR condition so test is valid regardless of Windows / Linux.

File: source/JavaPropertiesParser.Tests/ParserFixture.cs [hunk 9]
```diff
@@ -350,37 +351,49 @@ public void CanParseAValueWithALogicalNewLine()
...
         }
 
         [Test]
-        public void CanParseAValueWithAPhysicalNewLineAndIndentation()
+        [TestCase("value-with-physical-crlf-newline-and-indentation.properties")]
+        [TestCase("value-with-physical-lf-newline-and-indentation.properties")]
```
Comment (michaelongso) [inline source/JavaPropertiesParser.Tests/ParserFixture.cs:378]: As per above on CLRF and LF.

File: source/JavaPropertiesParser.Tests/ParserFixture.cs [hunk 10]
```diff
@@ -350,37 +351,49 @@ public void CanParseAValueWithALogicalNewLine()
...
+            {
+                Value("value", "val\\\r\n   ue"), // Windows newline - CRLF.
+                Value("value", "val\\\n   ue") // Linux & MacOS newline - LF. 
+            };
+
+            parsed.Expressions.Should().ContainSingle().Which.Should().BeAssignableTo<KeyValuePairExpression>();
+            var resultExpression = (KeyValuePairExpression)parsed.Expressions.First();
+            resultExpression.Key.Should().Be(expectedKey);
+            resultExpression.Separator.Should().Be(expectedSeparator);
+            expectedValues.Should().Contain(resultExpression.Value);
```
Comment (michaelongso) [inline source/JavaPropertiesParser.Tests/ParserFixture.cs:396]: As per above for `OR` condition

**Review by skermajo — APPROVED** [review]
👍 Thanks :) LGTM

**Pull request merged**